addons/mail/static/src/chatter/web/chatter_patch.js

```diff
@@ -305,6 +306,16 @@ patch(Chatter.prototype, {
         this.load(thread, ["followers", "suggestedRecipients"]);
     },
 
+    _onMounted() {
+        super._onMounted();
+        if (this.state.thread && this.props.highlightMessageId) {
+            this.state.thread.highlightMessage = {
+                id: this.props.highlightMessageId,
+                thread: this.state.thread,
+            };
```

Comment:
You can improve JS modeling to deduce `thread` of the highlightMessage when assigning in relation.
Indeed, if you assign `thread.highlightMessage`, the message is necessarily in `thread`. For example you could define
```js
highlightMessage = Record.one("Message", {
    onAdd(msg) {
        msg = this;
    },
});
```